tree/ntuple/v7/src/RField.cxx

```diff
@@ -1083,6 +1083,211 @@ void ROOT::Experimental::RVectorField::AcceptVisitor(Detail::RFieldVisitor &visi
 
 //------------------------------------------------------------------------------
 
+ROOT::Experimental::RRVecField::RRVecField(std::string_view fieldName, std::unique_ptr<Detail::RFieldBase> itemField)
+   : ROOT::Experimental::Detail::RFieldBase(fieldName, "ROOT::VecOps::RVec" + itemField->GetType() + ">",
+                                            ENTupleStructure::kCollection, false /* isSimple */),
+     fItemSize(itemField->GetValueSize()), fNWritten(0)
+{
+   Attach(std::move(itemField));
+}
+
+std::unique_ptr<ROOT::Experimental::Detail::RFieldBase>
+ROOT::Experimental::RRVecField::CloneImpl(std::string_view newName) const
+{
+   auto newItemField = fSubFields[0]->Clone(fSubFields[0]->GetName());
+   return std::make_unique<RRVecField>(newName, std::move(newItemField));
+}
+
+void ROOT::Experimental::RRVecField::AppendImpl(const Detail::RFieldValue &value)
+{
+   void **rvecAddr = reinterpret_cast<void**>(value.GetRawPtr());
+   // void* fBegin is the first data member
+   void *begin = *rvecAddr;
+   // int32_t fSize is the second data member (after 1 void*)
+   const int32_t size = *reinterpret_cast<int32_t *>(rvecAddr + 1);
+   R__ASSERT(size >= 0);
+
+   for (int32_t i = 0; i < size; ++i) {
+      auto elementValue = fSubFields[0]->CaptureValue(reinterpret_cast<void *>(std::intptr_t(begin) + i * fItemSize));
+      fSubFields[0]->Append(elementValue);
+   }
+
+   fColumns[0]->Append(Detail::RColumnElement<ClusterSize_t>(&fNWritten));
+   fNWritten += size;
+}
+
+void ROOT::Experimental::RRVecField::ReadGlobalImpl(NTupleSize_t globalIndex, Detail::RFieldValue *value)
+{
+   // TODO as a performance optimization, we could assign values to elements of the inline buffer
```

Comment:
I think that's an important optimization

Reply:
I agree! Do you want it in this PR or can I open a subsequent one with just this optimization?

Reply:
(will add in a later PR)